src/core.py

```
@@ -1,396 +1,1205 @@
+        whitelist_settings = settings.get("whitelist", {}) or {}
+
+        trusted_proxies = ParsedNetworkSet.from_entries(server_settings.get("trusted_proxies", []) or [], "trusted_proxies")
+        always_allowed = ParsedNetworkSet.from_entries(
+            security_settings.get("always_allowed_ips", []) or [],
+            "always_allowed_ips",
+        )
+        path_exclusions = PathExclusions.from_settings(security_settings)
+
+        api_keys_config = settings.get("api_keys", []) or []
+        if not api_keys_config:
+            raise ValueError("Configuration must contain at least one API key")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)